app/models/manageiq/providers/redhat/inventory/persister/infra_manager.rb

```diff
@@ -1,11 +1,189 @@
 class ManageIQ::Providers::Redhat::Inventory::Persister::InfraManager < ManageIQ::Providers::Redhat::Inventory::Persister
-  include ManageIQ::Providers::Redhat::Inventory::Persister::Definitions::InfraCollections
-
   def initialize_inventory_collections
-    initialize_infra_inventory_collections
+    add_collection(infra, :disks)
+    add_collection(infra, :ems_clusters, :secondary_refs => {:by_uid_ems => %i[uid_ems]})
+    add_collection(infra, :ems_folders)
+    add_collection(infra, :guest_devices)
+    add_collection(infra, :hardwares)
+    add_collection(infra, :vm_and_template_ems_custom_fields)
+    add_collection(infra, :hosts)
+    add_collection(infra, :host_guest_devices)
+    add_collection(infra, :host_hardwares)
+    add_collection(infra, :host_networks)
+    add_collection(infra, :host_operating_systems)
+    add_collection(infra, :host_storages)
+    add_collection(infra, :host_switches)
+    add_collection(infra, :host_virtual_switches)
+    add_collection(infra, :lans)
+    add_collection(infra, :networks)
+    add_collection(infra, :operating_systems)
+    add_collection(infra, :vms)
+
+    add_datacenters
+    add_miq_templates
+    add_resource_pools
+    add_snapshots
+    add_storages
+    add_ems_folder_children
+    add_ems_cluster_children
+  end
+
+  # group :ems_clusters
+  def add_resource_pools
+    add_collection(infra, :resource_pools) do |builder|
+      if targeted?
+        builder.add_targeted_arel(
+          lambda do |_inventory_collection|
+            manager.resource_pools.where(:uid_ems => references(:ems_clusters).collect { |ref| "#{URI(ref).path.split('/').last}_respool" })
+          end
+        )
+      end
+    end
+  end
+
+  def add_datacenters
+    add_collection(infra, :datacenters) do |builder|
+      builder.add_properties(:arel => manager.ems_folders.where(:type => 'Datacenter'))
+
+      if targeted?
+        builder.add_targeted_arel(
+          lambda do |_inventory_collection|
+            manager.ems_folders.where(:type => 'Datacenter').where(:ems_ref => references(:datacenters))
+          end
+        )
+      end
+    end
   end
 
-  def strategy
-    @collection_group == :datacenters ? :local_db_find_missing_references : nil
```

Comment:
@borod108 where was this used?  I only ever see it being set to `nil` in the [parent persister](https://github.com/ManageIQ/manageiq-providers-ovirt/pull/361/files#diff-b3e49d765d9e4c3ce0a3b2b1afaf954aL10)

Reply:
@agrare it was removed in #273. `@collection_group` variable was there because of there were different strategies for some ICs. @Ladas - do you know why did you remove it?

Reply:
I don't remember :-)